main/githooks.py

```diff
@@ -66,14 +81,36 @@ def _is_windows():
 
 def get_user():
     '''Get user making the commit'''
-    output = _get_output('git var GIT_AUTHOR_IDENT')
-    match = re.match(r'^(.+) <', output)
-    return match.group(1)
+    if _is_github_event():
+        return os.environ['GITHUB_ACTOR']
+    else:
+        output = _get_output('git var GIT_AUTHOR_IDENT')
+        match = re.match(r'^(.+) <', output)
+        return match.group(1)
 
 
 def get_branch():
     '''Get current branch'''
-    return _get_output('git branch').split()[-1]
+    if _is_github_event():
+        if _is_pull_request():
+            return os.environ['GITHUB_HEAD_REF']
+        else:
+            return os.environ['GITHUB_REF'].split('/')[-1]
+    else:
+        return _get_output('git branch').split()[-1]
+
+
+def get_sha():
+    '''Get the commit sha'''
```

Comment:
Is this available via the environment?

Reply:
$GITHUB_SHA returns the sha of the fake "merge commit", not the sha of the feature branch head.

Reply:
Fair enough, might be worth commenting on why this code uses a call to git instead of the environment variable so that when the next developer arrives they won't think it was done differently because (any reason other than 'it doesn't work that way') :)

Reply:
Good point.